uniffi_bindgen/src/interface/mod.rs

```diff
@@ -105,9 +105,18 @@ impl<'ci> ComponentInterface {
     pub fn from_webidl(idl: &str) -> Result<Self> {
         let mut ci = Self::default();
         ci.uniffi_version = env!("CARGO_PKG_VERSION").to_string();
-        // There's some lifetime thing with the errors returned from weedle::parse
+        // There's some lifetime thing with the errors returned from weedle::Definitions::parse
         // that my own lifetime is too short to worry about figuring out; unwrap and move on.
-        let defns = weedle::parse(idl.trim()).unwrap();
+
+        // Note we use `weedle::Definitions::parse` instead of `weedle::parse` so
+        // on parse errors we can see how far weedle got, which helps locate the problem.
+        use weedle::Parse; // this trait must be in scope for parse to work.
+        let (remaining, defns) = weedle::Definitions::parse(idl.trim()).unwrap();
+        if !remaining.is_empty() {
+            println!("Error parsing the IDL. Text remaining to be parsed is:");
+            println!("{}", remaining);
+            bail!("parse error");
+        }
```

Comment:
re 1: I'm not that bothered by this because the actual location of the error is usually a long way from what is reported - as I say in the first comment, this is only marginally better than the status quo, so trying to pretend it's more precise doesn't seem that useful.

re 2: The comments above seem to indicate that (I assume) Ryan had previously tried to avoid the `unwrap()` and failed, and given this isn't part of the "runtime" I see no reason to make myself suffer in the same way :)

So I think I'll just land this as is and we can consider other improvements later.